### DISCUSSION TITLE
Run locally without proxy

### DISCUSSION THREAD

**Discussion by ktech-davidp** [aws/graph-explorer#134] [Q&A] [answered]
How can I run locally without using an EC2 proxy? We have a site-to-site VPN set up, so a proxy is not necessary to connect to our neptune database. 
I can't run without a host, and when including neptune endpoint as a host it says `TypeError [ERR_INVALID_URL]: Invalid URL`

**Comment by triggan** [comment, marked as answer]
Firstly, do not be confused by running Graph Explorer on an EC2 instance and the concept of Graph Explorer's proxy.  Those are two distinct things.  Graph Explorer can run on an EC2 instance or on your local machine.  The Graph Explorer proxy is used for a couple of purposes - 1) as a proxy into the VPC where Neptune is located (if Graph Explorer is run on an EC2 instance or a container in the same VPC with Neptune - for users that do not have direct network connectivity into their VPC) and 2) as a proxy to sign requests going to Neptune if the Neptune cluster is configured with IAM Authentication enabled.

If you do not have IAM Auth enabled on your Neptune cluster, you can run the container on your local machine and connect to it using https://localhost/explorer without using the proxy configuration (be sure to add the /explorer, otherwise you'll see an error message).  If IAM Auth is enabled, then you'll need to pass IAM credentials into the container and use the proxy to sign the requests sent by Graph Explorer to Neptune.  More details in this recently closed issue:  https://github.com/aws/graph-explorer/issues/127
